app/scripts/modules/core/src/application/nav/NavRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { UIRouterContext } from '@uirouter/react-hybrid';
+import { UISref, UISrefActive } from '@uirouter/react';
+
+import { NavCategory } from './NavCategory';
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../application.model';
+
+export interface INavRouteProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+@UIRouterContext
+export class NavRoute extends React.Component<INavRouteProps> {
```

Comment:
If you're interested, it's possible to turn this into a functional component by [rendering a `<UIRouterContextComponent>` element](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/presentation/modal/showModal.tsx#L84-L86) at the root (vs. using the `@UIRouterContext` decorator)

Reply:
Very interested thanks! I appreciate the context with existing helper components in this PR.